File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;
class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        
        for(int n : priorities){
            pq.add(n);
        }
        while(!pq.isEmpty()) {
            for(int i = 0; i<priorities.length; i++){
                if(priorities[i] == pq.peek()){
                    pq.poll();
                    answer++;
                    if(location == i){
                        return answer;
                    }
                    
                }
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    answer = 0
    q = [(i, p) for i,p in enumerate(priorities)]
    while True:
        cur = q.pop(0)
        if any(cur[1] < e[1] for e in q):
            q.append(cur)
        else:
            answer+=1
            if cur[0] == location:
                return answer